api/server/sdk/volume_ops.go

```diff
@@ -623,11 +662,21 @@ func mergeVolumeSpecsPolicy(vol *api.VolumeSpec, req *api.VolumeSpecPolicy) *api
 	}
 
 	// HA Level
-	// validate if given HA Level is >= policyRepl level
-	if req.GetHaLevelOpt() != nil && req.GetHaLevel() >= vol.GetHaLevel() {
-		spec.HaLevel = req.GetHaLevel()
-	} else {
-		spec.HaLevel = vol.GetHaLevel()
+	spec.HaLevel = vol.GetHaLevel()
+	if req.GetHaLevelOpt() != nil {
+		switch req.GetHaLevelOperator() {
+		case api.VolumeSpecPolicy_Maximum:
+			if vol.GetHaLevel() > req.GetHaLevel() {
+				spec.HaLevel = req.GetHaLevel()
+			}
+		case api.VolumeSpecPolicy_Minimum:
+			if vol.GetHaLevel() < req.GetHaLevel() {
```

Comment:
shouldn't it keep HA level as it is if above spec policy min HA level
line no 627